amazon/ion/core.py

```diff
@@ -471,3 +474,87 @@ def timestamp(year, month=1, day=1,
         hour, minute, second, microsecond,
         tz, precision=precision, fractional_precision=fractional_precision
     )
+
+
+class Multimap(MutableMapping):
+    """
+    Dictionary that can hold multiple values for the same key
+
+    In order not to break existing customers, getting and inserting elements with ``[]`` keeps the same behaviour
+    as the built-in dict. If multiple elements are already mapped to the key, ``[]`  will return
+    the newest one.
+
+    To map multiple elements to a key, use the ``add_item`` operation.
+    To retrieve all the values map to a key, use ``get_all_values``.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(Multimap, self).__init__()
+        self.store = {}
```

Comment:
Should this be named `self.__store`, as it's not intended to be accessed publicly?

Reply:
Addressed